### ISSUE TITLE
Broken ghash for PowerPCISA on OpenSSL 1.0.2

### ISSUE THREAD

**Issue by gwlim** [openssl/openssl#214]
ghash build error on PowerPC target 

```
-DOPENSSL_BN_ASM_MONT -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DAES_ASM -DVPAES_ASM   -c -o gcm128.o gcm128.c
gcm128.c: In function 'CRYPTO_gcm128_init':
gcm128.c:875:22: error: 'gcm_ghash_4bit' undeclared (first use in this function)
         ctx->ghash = gcm_ghash_4bit;
                      ^
gcm128.c:875:22: note: each undeclared identifier is reported only once for each function it appears in
make[5]: *** [gcm128.o] Error 1
make[5]: Leaving directory `/home/user/gamma/build_dir/target-powerpc_8540_uClibc-0.9.33.2/openssl-1.0.2/crypto/modes'
make[4]: *** [subdirs] Error 1
make[4]: Leaving directory `/home/user/gamma/build_dir/target-powerpc_8540_uClibc-0.9.33.2/openssl-1.0.2/crypto'
make[3]: *** [build_crypto] Error 1
make[3]: Leaving directory `/home/user/gamma/build_dir/target-powerpc_8540_uClibc-0.9.33.2/openssl-1.0.2'
make[2]: *** [/home/user/gamma/build_dir/target-powerpc_8540_uClibc-0.9.33.2/openssl-1.0.2/.built] Error 2
make[2]: Leaving directory `/home/user/gamma/package/libs/openssl'
make[1]: *** [package/libs/openssl/compile] Error 2
```

Compiles fine if http://git.openssl.org/gitweb/?p=openssl.git;a=commit;h=6cd13f70bb51eb7982db8ab9f56c776b0c32b3ed 
is applied but I don't think that is right.

**Comment by gwlim** [comment]
It is a older processor e500v2 hence non PPC_CRYPTO207 architecture

**Comment by gwlim** [comment]
This fixes the build error but I am not sure if it is correct
```
--- a/crypto/modes/gcm128.c	2015-01-22 22:58:32.000000000 +0800
+++ b/crypto/modes/gcm128.c	2015-01-25 12:47:47.036810297 +0800
@@ -870,9 +870,9 @@ void CRYPTO_gcm128_init(GCM128_CONTEXT *
         ctx->gmult = gcm_gmult_p8;
         ctx->ghash = gcm_ghash_p8;
     } else {
-        gcm_init_4bit(ctx->Htable, ctx->H.u);
-        ctx->gmult = gcm_gmult_4bit;
-        ctx->ghash = gcm_ghash_4bit;
+        gcm_init_p8(ctx->Htable, ctx->H.u);
+        ctx->gmult = gcm_gmult_p8;
+        ctx->ghash = gcm_ghash_p8;
     }
 # else
     gcm_init_4bit(ctx->Htable, ctx->H.u);
```

**Comment by dot-asm** [comment]
This was addressed in b2991c081aba5351a3386bdde2927672d53e5c99

**Comment by gwlim** [comment]
Verified Fixed Thanks!
